serverlessworkflow/modules/ROOT/pages/use-cases/timeout-showcase-example.adoc

```diff
@@ -26,218 +33,254 @@ image::core/callback-state-timeouts.svg[]
 }
 ----
 
-The timeout is configured with a maximum time 30 seconds to be waited by the workflow to receive the callbackEvent, in case it does not arrive in time, the execution moves, and the eventData variable remains null.
-See the xref:core/timeouts-support.adoc#callback-state[callback state definition].
+.CallbackState definition
+[source,json]
+----
+{
+ "name": "CallbackState",
+ "type": "callback",
+ "action": {
+   "name": "callbackAction",
+   "functionRef": {
+     "refName": "callbackFunction",
+     "arguments": {
+       "input": "${\"callback-state-timeouts: \" + $WORKFLOW.instanceId + \" has executed the callbackFunction.\"}"
+     }
+   }
+ },
+ "eventRef": "callbackEvent",
+ "transition": "CheckEventArrival",
+ "onErrors": [
+   {
+     "errorRef": "callbackError",
+     "transition": "FinalizeWithError"
+   }
+ ],
+ "timeouts": {
+   "eventTimeout": "PT30S"
+ }
+}
+----
 
+The timeout is configured with a duration of 30 seconds, and if no event arrives during this time, the flow execution moves to the next state, and the workflow's data remains unchanged.
 
-== Switch workflow
+On the other hand, if the event arrive, the event payload is merged into the workflow's data, and thus, the `eventData` property of the workflow's data, will contain the information carried by the event payload.
+Using this simple configuration strategy, the workflow can collect the event information, and use it for example to determine the path to go in the next state.
 
-The switch example is similar to the callback but once the execution reaches the state, it waits for one of the two configured events, `visaDeniedEvent` or `visaApprovedEvent`, to arrive, see the <<switch-state, switch state definition>>.
+See the xref:core/timeouts-support.adoc#callback-state-timeout[callback state definition] for more information.
 
-If any of the configured events arrives before the timeout is overdue, the workflow execution moves to the next state defined in the corresponding `transition`.
+For more information about how the incoming event information can be merged into the workflow's data you can see: link:{spec_doc_url}#Event-data-filters[Event data filters].
 
-If none of the events arrive before the timeout is overdue, the workflow then moves to the state defined in `defaultCondition` transition.
+[#switch_state_timeouts_workflow]
+== `switch_state_timeouts` workflow
 
-.Switch timeout workflow
-image::core/switch-state-timeouts.svg[]
```

Comment:
Just now realized, are those moved images without the `-decorated` suffix used anywhere?

Reply:
It's not referenced directly by the document, but it's the original picture about how the workflow looked like. And it's kept like a backup copy of this image in case we need it for future changes/modifications where we might want to go back to the original shape. Facilitates the documentation maintaining/modification procedure. Same for the other shapes where we have xxx and the xxx-decorated.